control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -473,7 +478,7 @@ func (r *HostedControlPlaneReconciler) LookupReleaseImage(ctx context.Context, h
 	}
 	lookupCtx, lookupCancel := context.WithTimeout(ctx, 2*time.Minute)
 	defer lookupCancel()
-	return r.ReleaseProvider.Lookup(lookupCtx, hcp.Spec.ReleaseImage, pullSecret.Data[corev1.DockerConfigJsonKey])
+	return r.ReleaseProvider.Lookup(lookupCtx, hcp.Spec.ReleaseImage, pullSecret.Data[corev1.DockerConfigJsonKey], r.RegistryOverrides)
```

Comment:
Instead of changing the ReleaseProvider interface, I think it makes more sense to define a new implementation of the interface that wraps an inner provider and takes a set of registry overrides. That way all lookups stay the same and you only change where the ReleaseProvider is created. For examples of what I mean, see https://github.com/openshift/hypershift/blob/main/support/releaseinfo/cached_provider.go and https://github.com/openshift/hypershift/blob/main/support/releaseinfo/static_provider.go

Reply:
You did a great job of explaining this and I think this will be a lot cleaner. Making change now